controllers/auth_controllers.js

```diff
@@ -8,52 +8,101 @@ const schema = new passwordValidator();
+    .is().min(5)                                    // Minimum length 8
+    .is().max(100)                                  // Maximum length 100
+    .has().uppercase()                              // Must have uppercase letters
+    .has().lowercase()                              // Must have lowercase letters
+    .has().digits()                                //  any  digits allow 
 
 _exports.registerController = async (req, res) => {
-    try
-    {
+    try {
         const body = req.body;
         const email = body.email;
         const password = body.password;
         const firstName = body.firstName;
         const lastName = body.lastName;
-        if (!schema.validate(password) || !emailValidator.validate(email))
-        {return res.sendStatus(500)}
+        if (!schema.validate(password) || !emailValidator.validate(email)) { return res.sendStatus(500) }
 
         const hashedPassword = await bcrypt.hash(password, 10)
-       
+
         const new_user = new User({
             firstName,
             lastName,
-            email,  
+            email,
             password: hashedPassword
         });
         const user = await new_user.save();
-        const token = jwt.sign({userEmail: user.email}, process.env.secretKey, { 
+        const token = jwt.sign({ userEmail: user.email }, process.env.secretKey, {
             expiresIn: '15m'
         });
-        const refreshToken = jwt.sign({userEmail: user.email}, process.env.secretKey, { 
+        const refreshToken = jwt.sign({ userEmail: user.email }, process.env.secretKey, {
             expiresIn: '24h'
         });
-        res.cookie('token',token,{httpOnly:true,maxAge:1000*60*15})
-        res.cookie('refreshToken',refreshToken,{httpOnly:true,maxAge:1000*60*60*24})
-        user.refresh_tokens.push({token:refreshToken,expiration: new Date(
-            Date.now() + 1000*60*60*24
-        )})
+        res.cookie('token', token, { httpOnly: true, maxAge: 1000 * 60 * 15 })
+        res.cookie('refreshToken', refreshToken, { httpOnly: true, maxAge: 1000 * 60 * 60 * 24 })
+        user.refresh_tokens.push({
+            token: refreshToken, expiration: new Date(
+                Date.now() + 1000 * 60 * 60 * 24
+            )
+        })
         await user.save()
 
         console.log(req.body);
         res.status(200).json(user);
     }
-    catch(err)
-    {
+    catch (err) {
         console.log(err);
         res.sendStatus(500);
     }
 }
+
+_exports.loginController = async (req, res) => {
+    try {
+        const { email, password } = req.body
+        let user = await User.findOne({ email })
+
+        if (!password || !email) {
+            res.send({ message: "Invalid Credentials" })
+        }
+
+        if (!user) {
+            res.send({ error: "Invalid Credentials" });
+        }
+        const validPassword = await bcrypt.compare(password, user.password)
+
+        if (validPassword) {
```

Comment:
Use if password is not valid return error status. Otherwise route will never respond to request.